libs/portals/admin/regulations-admin/src/state/validations.ts

```diff
@@ -0,0 +1,313 @@
+import { HTMLText, PlainText } from '@island.is/regulations'
+import { makeHighAngstWarnings } from '@island.is/regulations-tools/useTextWarnings'
+import { errorMsgs } from '../lib/messages'
+import {
+  findAffectedRegulationsInText,
+  findSignatureInText,
+} from '../utils/guessers'
+import {
+  DraftField,
+  DraftImpactForm,
+  DraftingState,
+  HtmlDraftField,
+  RegDraftForm,
+  RegDraftFormSimpleProps,
+} from './types'
+
+const propMap: Record<RegDraftFormSimpleProps, true> = {
+  title: true,
+  text: true,
+  idealPublishDate: true,
+  fastTrack: true,
+  effectiveDate: true,
+  signatureDate: true,
+  signatureText: true,
+  signedDocumentUrl: true,
+  lawChapters: true,
+  ministry: true,
+  type: true,
+  draftingNotes: true,
+  authors: true,
+  name: true,
+}
+const draftRootProps = Object.keys(
+  propMap,
+) as ReadonlyArray<RegDraftFormSimpleProps>
+
+// ---------------------------------------------------------------------------
+
+export const validateImpact = (
+  impact: DraftImpactForm,
+  state?: DraftingState,
+) => {
+  // TODO: perform more validations, such as date boundry checks, etc.
```

Comment:
Todo!

Reply:
Yes! There is one crucial validation missing.